include/faabric/util/dirty.h

```diff
@@ -14,6 +16,21 @@
 #define PAGEMAP_ENTRY_BYTES sizeof(uint64_t)
 #define PAGEMAP_SOFT_DIRTY (1Ull << 55)
 
+// The following isn't available using HWE on 20.04, so copied from:
```

Comment:
There are a few constants in this file that are different (eg. UFFD_API_FEATURES lists all valid feature bits), it might be better to just copy in the entire userfaultfd.h file into the repo and include it above from the local copy rather than the system-wide one

Reply:
Hmm yep good point, my thinking was that if we use the system-wide version for everything we can, then we'll reduce the risk of diverging from the API (and some potentially very difficult-to-diagnose issues). However, it makes it much easier to diff and work out what's gone wrong if we have the whole file copied in, so I'll do that.